modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java

```diff
@@ -224,4 +252,67 @@ public int hashCode() {
             }
         }
     }
+
+    @Override
+    public List<PersistentTasksExecutor<?>> getPersistentTasksExecutor(ClusterService clusterService, ThreadPool threadPool,
+                                                                       Client client, SettingsModule settingsModule,
+                                                                       IndexNameExpressionResolver expressionResolver) {
+        return List.of(new GeoIpDownloader(client, new HttpClient(), clusterService, threadPool, settingsModule.getSettings()));
```

Comment:
I think for the time being let's add a feature flag here and only load this component if the feature flag is enabled?

Reply:
Maybe `geoip.downloader.enabled` setting's default value should just depend on the feature flag.

Reply:
Done